### PR TITLE
Add custom conditions use cases to the docs

### PR DESCRIPTION
**Pull request by myovchev**
Details  
- Add custom conditions use cases to the docs
- Add tests for corresponding examples
- Add version requirements for every use case in the documentation
- Propose unified terminology to avoid confusion:  
`core condition`: `AND`, `OR`. `NOT`, etc
`custom condition`: declared and registered custom condition `conditionName`, available in the grand  flow as `custom:conditionName`  
`function condition`: native JS function, added while defining the grands, not serializeable
- Expose `getValueByPath` from condition utils to allow developers to use JSONPath in their custom condition implementations. It's not exposed in the browser build (`./app.js`), but I'm not sure how it'll affect the bundle size.

Ref: #45

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,9 +20,10 @@ e.g. `ac.can(role).execute('create').on(resource)`
 - Grant permissions by attributes defined by glob notation (with nested object support).
 - Ability to filter data (model) instance by allowed attributes.
 - Ability to control access using conditions.
-- Supports AND, OR, NOT, EQUALS, NOT_EQUALS, STARTS_WITH, LIST_CONTAINS conditions.
-- You can specify dynamic context values in the conditions using JSON Paths.
-- You can define your own condition functions too but please note if you use custom functions instead of standard conditions, you won't be able to save them as json in the DB.
+- Supports AND, OR, NOT, EQUALS, NOT_EQUALS, STARTS_WITH, LIST_CONTAINS core conditions.
+- You can specify dynamic context values in the core conditions using JSON Paths.
+- Supports your own custom conditions e.g. `custom:isArticleOwner`.
+- You can define your own function conditions too but please note if you use custom functions instead of standard conditions, you won't be able to save them as json in the DB.
```
Comment (koladilip) [inline README.md:26]: This is no longer true, we are supporting storing of custom functions

**Review by koladilip — CHANGES_REQUESTED** [review]

**Comment by koladilip** [comment]
@myovchev  Thanks a lot for this change.

File: README.md (hunk 1 above)
Comment (myovchev) [reply README.md:26]: This is what the old README says. Can you elaborate on this please? How a pure JS function is serialized?
I can easily remove the line, but there is one more mention of that below, so I need to understand what it is all about.

Comment (koladilip) [reply README.md:26]: https://github.com/tensult/role-acl/blob/develop/src/utils/common.ts#L36
Here is the small function I am using for the same.

**Review by koladilip — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (myovchev) [reply README.md:26]: Good to know! Not my "style" but it's a neat feature :)

**Pull request merged**